rpcserver.go

```diff
@@ -2993,6 +2995,109 @@ func (r *rpcServer) QueryAssetRoots(ctx context.Context,
 		return nil, err
 	}
 
+	// If no root was found, the asset may be a grouped asset. Check if we
+	// have a group stored that contains this asset, and return the roots
+	// for that asset group if so.
+	mayBeGrouped := errors.Is(issuanceErr, universe.ErrNoUniverseRoot) &&
+		errors.Is(transferErr, universe.ErrNoUniverseRoot) &&
+		universeID.GroupKey == nil
+
+	var (
+		groupedAssetID asset.ID
+		foundGroupKey  *btcec.PublicKey
+	)
+
+	// Attempt to look up a matching asset group.
+	if mayBeGrouped {
+		groupedAssetID = universeID.AssetID
+		rpcsLog.Debugf("No roots found for asset %v, checking if this "+
+			"asset is part of a group", groupedAssetID.String())
+
+		assetGroup, err := r.cfg.TapAddrBook.QueryAssetGroup(
```

Comment:
Doing it at this level means another round trip (actually 3x) and set of DB queries each time this path is exercised. Alternatively, we can do this from within the main DB query, as we have the asset group view available. Doing it at that lower level woudl also mean we can test this more effectively, which is something we generally want to improve on now that we have the initial alpha release out.

Reply:
Discussed offline that we'll defer the optimization to a future PR, but clean up the call site a bit here with some helper functions.